File: backend/admin-center/src/main/java/com/admin/entity/Department.java
package com.admin.entity;

import com.admin.enums.DepartmentStatus;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

/**
 * 部门实体
 */
@Entity
@Table(name = "admin_departments")
@EntityListeners(AuditingEntityListener.class)
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class Department {
    
    @Id
    @Column(length = 64)
    private String id;
    
    @Column(name = "name", nullable = false, length = 100)
    private String name;
    
    @Column(name = "code", nullable = false, unique = true, length = 50)
    private String code;
    
    @Column(name = "parent_id", length = 64)
    private String parentId;
    
    @Column(name = "level", nullable = false)
    @Builder.Default
    private Integer level = 1;
    
    @Column(name = "path", length = 500)
    private String path;
    
    @Column(name = "manager_id", length = 64)
    private String managerId;
    
    @Column(name = "phone", length = 50)
    private String phone;
    
    @Column(name = "description", columnDefinition = "TEXT")
    private String description;
    
    @Column(name = "cost_center", length = 50)
    private String costCenter;
    
    @Column(name = "location", length = 200)
    private String location;
    
    @Column(name = "sort_order")
    @Builder.Default
    private Integer sortOrder = 0;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    @Builder.Default
    private DepartmentStatus status = DepartmentStatus.ACTIVE;
    
    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private Instant createdAt;
    
    @CreatedBy
    @Column(name = "created_by", length = 64, updatable = false)
    private String createdBy;
    
    @LastModifiedDate
    @Column(name = "updated_at")
    private Instant updatedAt;
    
    @LastModifiedBy
    @Column(name = "updated_by", length = 64)
    private String updatedBy;
    
    @Transient
    @Builder.Default
    private List<Department> children = new ArrayList<>();
    
    @Transient
    private Long memberCount;
    
    /**
     * 检查是否是根部门
     */
    public boolean isRoot() {
        return parentId == null || parentId.isEmpty();
    }
    
    /**
     * 检查是否是指定部门的祖先
     */
    public boolean isAncestorOf(Department other) {
        if (other == null || other.getPath() == null) return false;
        return other.getPath().startsWith(this.path + "/");
    }
    
    /**
     * 检查是否是指定部门的后代
     */
    public boolean isDescendantOf(Department other) {
        if (other == null || this.path == null) return false;
        return this.path.startsWith(other.getPath() + "/");
    }
}

File: frontend/admin-center/src/api/department.ts
import { get, post, put, del } from './request'

export interface Department {
  id: string
  name: string
  code: string
  parentId?: string
  level: number
  path?: string
  managerId?: string
  managerName?: string
  phone?: string
  description?: string
  costCenter?: string
  location?: string
  sortOrder: number
  status: 'ACTIVE' | 'INACTIVE'
  memberCount?: number
  createdAt: string
  updatedAt: string
  children?: Department[]
}

export interface CreateDepartmentRequest {
  name: string
  code: string
  parentId?: string
  managerId?: string
  phone?: string
  description?: string
  costCenter?: string
  location?: string
  sortOrder?: number
}

export interface UpdateDepartmentRequest {
  name?: string
  managerId?: string
  phone?: string
  description?: string
  costCenter?: string
  location?: string
  sortOrder?: number
}

export const departmentApi = {
  // 获取部门树
  getTree: () => get<Department[]>('/departments/tree'),
  
  // 获取部门列表
  list: (params?: { parentId?: string; status?: string }) => 
    get<Department[]>('/departments', { params }),
  
  // 根据ID获取部门
  getById: (id: string) => get<Department>(`/departments/${id}`),
  
  // 创建部门
  create: (data: CreateDepartmentRequest) => post<Department>('/departments', data),
  
  // 更新部门
  update: (id: string, data: UpdateDepartmentRequest) => put<Department>(`/departments/${id}`, data),
  
  // 删除部门
  delete: (id: string) => del<void>(`/departments/${id}`),
  
  // 获取部门成员
  getMembers: (id: string, params?: { page?: number; size?: number }) => 
    get<any>(`/departments/${id}/members`, { params }),
  
  // 获取子部门
  getChildren: (id: string) => get<Department[]>(`/departments/${id}/children`),
  
  // 获取部门路径（从根到当前部门）
  getPath: (id: string) => get<Department[]>(`/departments/${id}/path`)
}
